### PR TITLE
CI:  create separate public PR YAML

### PR DESCRIPTION
**Pull request by dtivel**
Progress on https://github.com/dotnet/sign/issues/676.

The PR copies /.vsts-ci.yml to ./.vsts-pr.yml then adds a comment to the top of each file indicating the AzDO pipeline URL.

### CODE REVIEW

File: .vsts-pr.yml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    value: ${{ eq(variables['System.TeamProject'], 'internal') }}
+  - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+    - group: DotNet-Sign-SDLValidation-Params
+
+# CI and PR triggers
+trigger:
+  batch: true
+  branches:
+    include:
+    - main
```
Comment (jeffkl) [inline .vsts-pr.yml:20]: In the past, I've been bitten by this if I am updating a servicing branch.  If this repo will use servicing branches, I recommend an entry for it like `rel/*` or `release-*`.  Obviously if this tool will never be servicing older versions, its not needed.

File: .vsts-pr.yml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+              mergeTestResults: true
+              searchFolder: $(System.DefaultWorkingDirectory)
+              testRunTitle: sign unit tests - $(Agent.JobName)
+            condition: succeededOrFailed()
+          - task: ComponentGovernanceComponentDetection@0
+            displayName: Component Governance scan
+            inputs:
+              ignoreDirectories: '$(Build.SourcesDirectory)/.packages,$(Build.SourcesDirectory)/artifacts/obj/Sign.Cli'
+
+- ${{ if eq(variables['System.TeamProject'], 'internal') }}:
```
Comment (jeffkl) [inline .vsts-pr.yml:126]: Will this condition ever be true? I wonder if these make the pipeline more complex for no reason

File: .vsts-pr.yml (hunk 1 above)
Comment (dtivel) [reply .vsts-pr.yml:20]: Great feedback.  Currently, we don't have release branches.  We've only had prereleases.  We'll cross that bridge when we get there.

File: .vsts-pr.yml (hunk 2 above)
Comment (dtivel) [reply .vsts-pr.yml:126]: Yes, in the internal pipeline.

After this merges, I will create a new PR to simplify this file by removing official-only stuff.

**Review by jeffkl — APPROVED** [review]

**Pull request merged**